source/Calamari/Deployment/PackageRetention/Model/PackageLocks.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using Calamari.Common.Plumbing.Deployment.PackageRetention;
+using Newtonsoft.Json;
+
+namespace Calamari.Deployment.PackageRetention.Model
+{
+    public class PackageLocks
+    {
+        [JsonProperty]
+        readonly Dictionary<ServerTaskID, DateTime> packageLocks;
+
+        [JsonConstructor]
+        internal PackageLocks(Dictionary<ServerTaskID, DateTime> packageLocks)
+        {
+            this.packageLocks = packageLocks ?? new Dictionary<ServerTaskID, DateTime>();
+        }
+
+        public PackageLocks() : this(new Dictionary<ServerTaskID, DateTime>())
+        {
+        }
+
+        public void AddLock(ServerTaskID deploymentTaskID)
+        {
+            if (packageLocks.ContainsKey(deploymentTaskID))
```

Comment:
Think we can just `packageLocks[deploymentTaskID] = DateTime.Now;` instead of the conditions

Reply:
Yep, thanks 👍